xds/internal/balancer/cdsbalancer/cdsbalancer.go

```diff
@@ -408,9 +409,6 @@ func (b *cdsBalancer) run() {
 				b.edsLB.Close()
 				b.edsLB = nil
 			}
-			if newXDSClient != nil {
```

Comment:
Is this an intended change, or merge conflict?

Reply:
I think I know why this change is here.

I would say don't make this change, or at least, don't mix it in this XL PR.

And it's an incomplete test cleanup. It leaves the balancers in an inconsistent state (if the balancer creates the client, it should call close, even though we know close is a noop)

And it only changes the easy part, but leaves the hard part behind. We should modify the tests to use attributes to pass the fake xDS client, so we can completely get rid of `newXDSClient`. But that change requires changing signature of `xds.SetClient()` to take an interface instead of `*Client`, so it can be the `fakeClient`.
I will make this change, to fix the tests in my other PR.

Reply:
I still think this doesn't give us enough benefit, to justify itself.
At least should not be mixed in a 84-file (trivial) change PR.
This diff really stick out when I was scanning the changes.

Reply:
#4510